packages/expo-updates/android/src/main/java/expo/modules/updates/loader/FileDownloader.java

```diff
@@ -190,6 +203,31 @@ public void onResponse(Call call, Response response) throws IOException {
     });
   }
 
+  private static JSONObject extractManifest(String manifestString, UpdatesConfiguration configuration) throws IOException {
+    try {
+      return new JSONObject(manifestString);
+    } catch (JSONException e) {
+      // Ignore this error, try to parse manifest as array
+    }
+
+    // TODO: either add support for runtimeVersion or deprecate multi-manifests
+    try {
+      // the manifestString could be an array of manifest objects
+      // in this case, we choose the first compatible manifest in the array
+      JSONArray manifestArray = new JSONArray(manifestString);
+      for (int i = 0; i < manifestArray.length(); i++) {
+        JSONObject manifestCandidate = manifestArray.getJSONObject(i);
+        String sdkVersion = manifestCandidate.getString("sdkVersion");
+        if (configuration.getSdkVersion() != null && Arrays.asList(configuration.getSdkVersion().split(",")).contains(sdkVersion)){
```

Comment:
Does the old code trim each of the SDK versions in the comma-separated list? If so, we should keep that behavior. If not, I think we can leave this as-is for now. (Same comment for the iOS implementation.)

Reply:
The old code uses pre-existing lists in the client codebase (`Constants.SDK_VERSION_LIST` on Android and `[EXVersions supportsVersion:]` on iOS) rather than splitting the string into a list itself. It does not do any trimming.